.gitlab-ci.yml

```diff
@@ -112,6 +112,136 @@ build:mender-client-acceptance-testing:
     paths:
       - checksums
 
+.template:build:mender-dist-packages-image:
+  stage: build
+  needs: []
+  image: docker:git
+  tags:
+    - mender-qa-worker-generic
+  services:
+    - docker:dind
+  before_script:
+    - apk --update --no-cache add bash curl aws-cli
+    - eval "$(curl https://raw.githubusercontent.com/mendersoftware/mendertesting/master/mender-ci-common.sh)"
+  script:
+    # GPG key is available only in protected branches
+    - if [ "$GPG_PRIV_KEY_BUILD" != "" ]; then
+    -   GPG_KEY_BUILD="$GPG_PRIV_KEY_BUILD"
+        ./mender-dist-packages-building/docker-build-image $DISTRO $RELEASE $ARCH
+    - else
+    -   ./mender-dist-packages-building/docker-build-image $DISTRO $RELEASE $ARCH
+    - fi
+    - docker save $DOCKER_REPOSITORY:mender-dist-packages-builder-$DISTRO-$RELEASE-$ARCH -o mender-dist-packages-builder-$DISTRO-$RELEASE-$ARCH.tar
+    - mender_ci_save_tmp_artifact mender-dist-packages-builder-$DISTRO-$RELEASE-$ARCH.tar
+  artifacts:
+    paths:
+      - checksums
```

Comment:
q: do you think we should save the images as job artifacts?

Reply:
It is a requirement for using the temporary S3 bucket. See [the documentation](https://github.com/mendersoftware/mendertesting/blob/master/mender-ci-common.sh#L31).